yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/internal/bytecode/types/gambit/ExpressionHandler.java

```diff
@@ -887,12 +907,27 @@ public void generate(CodeEmitter code) {
         private final Map<String, BytecodeExpression> fieldSettings = Maps.newLinkedHashMap();
         private final StructBuilder staticStructBuilder = createStruct();
 
+        private void initDynamicBuilder(Map<String, BytecodeExpression> fieldSettings) {
+            // reset and convert ourselves to dynamic
+            dynamic = true;
+            dynamicBuilder = new DynamicExpressionRecordBuilder();
+            // merge all of our existing properties to the dynamic builder
+            for (Map.Entry<String, BytecodeExpression> entry:fieldSettings.entrySet()) {
+                dynamicBuilder.add(Location.NONE, entry.getKey(), entry.getValue());
+            }              
+        }
+        
         @Override
         public RecordBuilder add(Location loc, String fieldName, BytecodeExpression input) {
             if(dynamic) {
                 dynamicBuilder.add(loc, fieldName, input);
                 return this;
             }
+            if (!isJavaFieldName(fieldName)) {
+                initDynamicBuilder(fieldSettings);
```

Comment:
Since initDynamicBuilder also sets the dynamic field, I think it would be fine for it to read the fieldSettings from the field rather than as a parameter.